Differences between VBA and Office Scripts (#93)
OfficeDev/office-scripts-docs: docs/resources/add-ins-differences.md

---
title: 'Differences between Office Scripts and Office Add-ins'
description: 'The behavior and API differences between Office Scripts and Office Add-ins.'
ms.date: 03/23/2020
localization_priority: Normal
---

# Differences between Office Scripts and Office Add-ins

Office Add-ins and Office Scripts have a lot in common. They both offer automated control of an Excel workbook through the `Excel` namespace of the Office JavaScript API. However, Office Scripts are more limited in their scope.

> ![A four-quadrant diagram showing the areas of focus for different Office extensibility solutions. Both Office Scripts and Office Web Add-ins are focused on web and collaboration, but Office Scripts are more end-user approachable (whereas Office Web Add-ins are targeted at a professional developer audience).)](../images/office-programmability-diagram.png)

Reviewer: ```suggestion
![A four-quadrant diagram showing the focus areas for different Office extensibility solutions. Both Office Scripts and Office Web Add-ins are focused on the web and collaboration, but Office Scripts cater to end users (whereas Office Web Add-ins target professional developers).)](../images/office-programmability-diagram.png)
```

Office Add-ins and Office Scripts have a lot in common. They both offer automated control of an Excel workbook through the `Excel` namespace of the Office JavaScript API. However, Office Scripts are more limited in their scope.

![A four-quadrant diagram showing the areas of focus for different Office extensibility solutions. Both Office Scripts and Office Web Add-ins are focused on web and collaboration, but Office Scripts are more end-user approachable (whereas Office Web Add-ins are targeted at a professional developer audience).)](../images/office-prgrammability-diagram.png)

> Office Scripts run to completion with a manual button press or as a step in [Power Automate](https://flow.microsoft.com/), whereas Office Add-ins rely on user interaction and persist while the workbook is in use. If you find that your Excel extension needs to exceed the scripting platform's capabilities, visit the [Office Add-ins documentation](/office/dev/add-ins) to learn more about Office Add-ins.

Reviewer 2: "persist" may not be clear - I had to think about what you meant for a second. It's also more correct to say the persistence is for the lifetime of the task pane. Maybe something like "OS doesn't maintain state between runs, but add-ins can" ?
